server/src/main/java/io/deephaven/server/arrow/ArrowFlightUtil.java

```diff
@@ -683,7 +702,10 @@ private synchronized void onExportResolved(final SessionState.ExportObject<Objec
                     minUpdateIntervalMs = options.minUpdateIntervalMs();
                 }
 
-                final Object export = parent.get();
+                Object export = parent.get();
+                if (export instanceof UncoalescedTable) {
+                    export = ((UncoalescedTable<?>) export).coalesce();
+                }
                 if (export instanceof QueryTable) {
                     final QueryTable table = (QueryTable) export;
```

Comment:
You could also do the same type validation on the result, but right now every `coalesce()` will give you a `QueryTable`. Same for all the `BaseTable` checks elsewhere.
```suggestion
                if (export instanceof Table) {
                    final QueryTable table = (QueryTable) ((Table) export).coalesce();
```